### PR TITLE
Add `load_many` to `DataLoader`

### PR DESCRIPTION
**Pull request by silas**
Add `load_many` to `DataLoader`.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

This release adds `load_many` to `DataLoader`.


---
Here's the preview release card for twitter:




Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Silas Sewell for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

File: strawberry/dataloader.py [hunk 1]
```diff
@@ -78,6 +88,9 @@ def load(self, key: K) -> Awaitable[T]:
 
         return future
 
+    def load_many(self, keys: Iterable[K]) -> Awaitable[List[T]]:
+        return gather(*[self.load(key) for key in keys])
+
```
Comment (BryceBeagle) [inline strawberry/dataloader.py:93]: Minor, nitpicky suggestion:
```suggestion
    def load_many(self, keys: Iterable[K]) -> Awaitable[List[T]]:
        return gather(*map(self.load, keys))
```

**Review by BryceBeagle — COMMENTED** [review]
Would you be able to add this to the [dataloader docs](https://strawberry.rocks/docs/guides/dataloaders)?

File: strawberry/dataloader.py (hunk 1 above)
Comment (silas) [reply strawberry/dataloader.py]: Updated

**Comment by silas** [comment]
@BryceBeagle let me know if that doc change works for you or if you'd like to see something else.

File: docs/guides/dataloaders.md [hunk 2]
```diff
@@ -74,6 +74,14 @@ await loader.load(1)
 
 Will result in only one call to `load_users`.
 
+To load multiple values in a single call we can use the `load_many`
+function, which has the same interface as `load_fn`, but uses the
+the DataLoader cache.
```
Comment (BryceBeagle) [inline docs/guides/dataloaders.md:79]: What do you mean by "has the same interface"? Do you mean the same signature? If so, they're slightly different (one takes a `List[]` and the other an `Iterable[]`.

**Review by BryceBeagle — CHANGES_REQUESTED** [review]

File: docs/guides/dataloaders.md (hunk 2 above)
Comment (silas) [reply docs/guides/dataloaders.md]: Yeah, I meant signature. Anyway I tweaked the wording.

**Review by BryceBeagle — APPROVED** [review]

**Comment by BryceBeagle** [comment]
btw, for future reference, we use the squash merge technique so there's no reason to keep the feature branch atomic. It makes it easier to re-review if you don't re-write history.

**Review by patrick91 — APPROVED** [review]
Nice one! Thank you <3

**Pull request merged**